src/api/ledger/parse/settings.js

```diff
@@ -2,27 +2,69 @@
 'use strict';
 const _ = require('lodash');
 const assert = require('assert');
-const AccountSetFlags = require('./utils').constants.AccountSetFlags;
+const AccountFlags = require('./utils').constants.AccountFlags;
+const AccountFlagIndices = require('./utils').constants.AccountFlagIndices;
 const parseFields = require('./fields');
 
 function getName(flagNumber) {
-  return _.findKey(AccountSetFlags, (v) => v === flagNumber);
+  return _.findKey(AccountFlagIndices, (v) => v === flagNumber);
 }
 
-function parseSettings(tx: Object) {
-  const txType = tx.TransactionType;
-  assert(txType === 'AccountSet' || txType === 'SetRegularKey');
+function parseFlags(tx: Object) {
   const settings = {};
+  if (tx.TransactionType !== 'AccountSet') {
+    return settings;
+  }
+
+  const modifiedNodes = tx.meta.AffectedNodes.filter(node =>
+    node.ModifiedNode.LedgerEntryType === 'AccountRoot');
+  assert(modifiedNodes.length === 1);
+  const node = modifiedNodes[0].ModifiedNode;
+
+  const oldFlags = _.get(node.PreviousFields, 'Flags');
```

Comment:
Sweet, I kind of came to like the `?` operator in coffee-script. I saw they added it to C# recently. 

`oldFlags = node.PreviousFields?.Flags`